indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -844,6 +858,48 @@ private void removeWorker(final Worker worker)
     }
   }
 
+  private void scheduleTasksCleanupForWorker(final String worker, final List<String> tasksToFail)
+  {
+    removedWorkerCleanups.put(
+        worker, exec.schedule(
+            new Runnable()
+            {
+              @Override
+              public void run()
+              {
+                log.info("Running scheduled cleanup for Worker[%s]", worker);
+                for (String assignedTask : tasksToFail) {
+                  String taskPath = JOINER.join(indexerZkConfig.getTasksPath(), worker, assignedTask);
+                  String statusPath = JOINER.join(indexerZkConfig.getStatusPath(), worker, assignedTask);
```

Comment:
Ultimately going ZkPaths instead of our own custom JOINER would be nice, but since this class uses JOINER everywhere, may as well keep with that for now.